### PR TITLE
chore(python): add .github/CODEOWNERS as a templated file

### PR DESCRIPTION
**Pull request by parthea**
I tested these changes by running the following commands after cloning synthtool
```
git checkout template-codeowners
docker build -t testcodeowners -f docker/owlbot/python/Dockerfile .
```

Once the docker image was built, I ran the following command in the root of python-contact-center-insights
```
docker run --user $(id -u):$(id -g) --rm -v $(pwd):/repo -w /repo template-codeowners
```

### CODE REVIEW

**Comment by tseaver** [comment]
@parthea Please do not merge this until checking that all "manual" libraries exclude `.github/CODEOWNERS` in their `owlbot.py`:  this PR would have negative impact on them, and make the discoverability / ease-of-customization for those repose unnecessarily harder.

**Comment by parthea** [comment]
@tseaver ,  I agree. I've added this as a requirement.

**Comment by busunkim96** [comment]
Was the check for `GAPIC_AUTO` removed intentionally?

**Comment by tseaver** [comment]
@busunkim96
> Was the check for `GAPIC_AUTO` removed intentionally?

I think we need to exclude `.github/CODEOWNERS` from being copied altogeher for the manual repos.  They have local policies which no amount of templating / `s.replace`-ing will not be worse than just excluding.

**Comment by parthea** [comment]
Capturing comment from https://github.com/googleapis/python-data-qna/pull/83 to be incorporated in this PR: 
> Could the synthtool template for CODEOWNERS note that the file is generated and that edits should be made to .repo-metadata.json?

File: synthtool/gcp/templates/python_library/.github/CODEOWNERS
Comment (tswast) [inline synthtool/gcp/templates/python_library/.github/CODEOWNERS:7]: Looks like the BigQuery repos are missing this property. Are you planning on sending some PRs to update the repo metadata?

Comment (parthea) [reply synthtool/gcp/templates/python_library/.github/CODEOWNERS:7]: Yes, I also need to add `default_version` to `.repo-metadata.json` for #1114. I'm planning to add it at the same time. I see nodejs already uses `default_version` so I'm following their lead. https://github.com/googleapis/nodejs-iap/blob/main/.repo-metadata.json#L12

Comment (parthea) [reply synthtool/gcp/templates/python_library/.github/CODEOWNERS:7]: Done.

**Comment by parthea** [comment]
@busunkim96 @tswast @tseaver This is ready for review

**Review by tswast — APPROVED** [review]

File: synthtool/gcp/templates/python_library/.github/CODEOWNERS
Comment (tswast) [inline synthtool/gcp/templates/python_library/.github/CODEOWNERS:1]: Should probably have a comment indicating that one needs to edit the repo metadata and that this is templated.

**Review by tswast — CHANGES_REQUESTED** [review]

File: synthtool/gcp/templates/python_library/.github/CODEOWNERS
Comment (parthea) [reply synthtool/gcp/templates/python_library/.github/CODEOWNERS:1]: Done

**Review by tswast — APPROVED** [review]

**Pull request merged**
